ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```diff
@@ -0,0 +1,186 @@
+        force: yes
+        backup: yes
+      become: true
+    
+    
+    - name: Update the CA trust files
+      command: update-ca-trust extract
+      become: true
+    
+    - name: Create container to serve the registry
+      shell: |
+        [[ $( podman ps -a --format '{{podman_format_name | quote }}' --filter name={{ pod_name_registry }} ) ]] \
+          && {
+               podman container stop {{ pod_name_registry }}
+               podman container rm {{ pod_name_registry }}
+             }
+    
+        podman create \
+          --name {{ pod_name_registry }} \
+          -p {{ registry_port_container }}:{{ registry_port }} \
+          -v {{ registry_dir_data }}:/var/lib/registry:z \
+          -v {{ registry_dir_auth }}:/auth:z \
+          -e "REGISTRY_AUTH=htpasswd" \
+          -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry" \
+          -e "REGISTRY_HTTP_SECRET=ALongRandomSecretForRegistry" \
+          -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+          -v {{ registry_dir_cert }}:/certs:z \
+          -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+          -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key docker.io/library/registry:2
+      become: true
+    
+    - name: Start container
+      command: podman start {{ pod_name_registry }}
+      become: true
+    
+    - name: Mirror remote registry to local
+      command:
+        oc adm release mirror
+          -a "{{ homedir }}/pullsecret.txt"
+          --from={{ release_image | quote }}
+          --to-release-image={{ local_registry | quote }}/{{ local_repo | quote }}:{{ release_version | quote }}
+          --to={{ local_registry | quote }}/{{ local_repo | quote }}
+    
+    - name: Read in the contents of domain.crt
+      slurp:
+        src: "{{ registry_dir_cert }}/domain.crt"
+      register: domain_cert_b64
+    
+    - name: Set trustbundle fact to contents of domain.crt
+      set_fact:
+        trustbundle: "{{ domain_cert_b64.content | string | b64decode }}"
+    
+    - name: Create mirrorred registry information
+      set_fact:
+        install_config_appends: "{{ lookup('template', 'install-config-appends.j2'
+    ) }}"
+    
+    - name: Create {{ install_config_appends_file }}
+      copy:
+        content: "{{ install_config_appends }}"
+        dest: "{{ homedir }}/{{ install_config_appends_file }}"
+    
+    - name: Create {{ install_config_appends_file }} on localhost
+      copy:
+        content: "{{ install_config_appends }}"
+        dest: "~/{{ install_config_appends_file }}"
+      delegate_to: localhost
+    
+    - name: Information
+      debug:
+        msg:
+        - "To use this mirrored registry to deploy you must do the following"
+        - "Add the authentication from either "
+        - "    {{ homedir }}/{{ registry_auth_file }} on {{ inventory_hostname }}"
+        - "    or ~/{{ registry_auth_file }} on this server to your pull secret."
+        - ""
+        - "Append the contents of either of the "
```

Comment:
It looks like this is done automatically by the playbook in the `install-config.j2` template.

Reply:
if you plan to reuse a registry without recreating it, you need the information in the files that this mentions.
Maybe changing the earlier line from reading: "To use this mirrored registry to deploy you must do the following" to "To reuse this mirrored registry for other deployments, you must do the following:"

Reply:
Ah, I see.  Thank you for the clarification!  But yeah, I was just confused at first and thinking that I needed to quickly modify the install-config.yaml that we generate before it was passed to openshift-install.

Reply:
the disconnected pieces set the pull secret to include the auth for the disconnected server before the install-config.yaml is created. I also modified the template to include the mirror information if it is set. I tried to make it just easily plug into the install-config.yaml process.

Reply:
changed message displayed to better reflect the intent of this.
pushed